### PR TITLE
Add WinnerIndex to OverallType

### PR DESCRIPTION
**Pull request by a-sink-a-wait**
Mainly looking for feedback; am I on the right track here? Don't want to commit any more time if I am way off base.

I've implemented a `winnerIndex` property from my understanding of just digging through the code. This is in an effort to display the winner in the replays folder of the desktop application.

### CODE REVIEW

File: src/stats/overall.ts [hunk 1]
```diff
@@ -67,6 +69,23 @@ export function generateOverallStats(
   return overall;
 }
 
+function determineWinner(playerStocks: StockType[], opponentStocks: StockType[]) : number {
+
+  if (playerStocks.length == 0 && opponentStocks.length == 0) return -1; //unknown winner;
```
Comment (jwsharpe) [inline src/stats/overall.ts:74]: can we avoid returning undefined by taking out the if statement and moving the `return -1` to the end of the function, will avoid weird edge cases or missing data errors later.

Comment (a-sink-a-wait) [reply src/stats/overall.ts:74]: Sounds good!

**Comment by a-sink-a-wait** [comment]
Updated this. Still need tests and some cleanup. Different approach, need some feeback.

**Comment by vinceau** [comment]
For starters, I'm not really sure I agree with having the winner index as a part of the overall stats object. Since it's not really a stat. 

But some comments on the code itself: I think it would be better to export a helper function that gets imported instead of having the player determination code in the `SlippiGame` class directly. Additionally you could probably use a reduce instead of a `forEach` [see here for inspiration](https://github.com/vinceau/slp-realtime/blob/master/src/utils/winner.ts).

The key issue with adding the winner (I assume for the purpose of displaying in the desktop app) is that it requires processing the entire file for all the frames `getStats()` which is not performant. People have already complained about the desktop app being slow already and at the moment, it only loads the metadata. This is the primary reason why winner is not included at the moment.

**Comment by a-sink-a-wait** [comment]
@vinceau Solid feedback. A `reduce` is definitely possible. I wanted something working and reviewable.

Is this worth implementing at the moment? Should we hold off?

**Comment by vinceau** [comment]
@a-sink-a-wait I think it would be best to hold off for the time being. Fizzi has talked about adding caching of sorts and pagination which would allow processing entire SLP files less costly. But until then, even if this was to be merged you probably wouldn't see it's use in the desktop app because the performance hit would be too costly.

**Comment by a-sink-a-wait** [comment]
I know there's this PR as well that aims to improve performance of the desktop App.
Would it be worth trying to combine these two implementations?

**Comment by NikhilNarayana** [comment]
> I know there's this PR as well that aims to improve performance of the desktop App. [project-slippi/slippi-desktop-app#58](https://github.com/project-slippi/slippi-desktop-app/pull/58)
> 
> Would it be worth trying to combine these two implementations?

It would be worth know what kind of performance we would be getting if combined.

**Comment by a-sink-a-wait** [comment]
I'll take a look into this.

**Comment by NikhilNarayana** [comment]
closing due to placements being in game end now
